docs/loading/Flink-connector-starrocks.md

```diff
@@ -1,173 +1,647 @@
+    
+              public RowData(int id, String name, int score) {
+                  this.id = id;
+                  this.name = name;
+                  this.score = score;
+              }
+          }
+      ```
+
+  - The main program is
+
+    ```java
+    // Generate records which use RowData as the container.
+    RowData[] records = new RowData[]{
+            new RowData(1, "starrocks-rowdata", 100),
+            new RowData(2, "flink-rowdata", 100),
+        };
+    DataStream<RowData> source = env.fromElements(records);
+
+    // Configure the connector with the required properties
+    StarRocksSinkOptions options = StarRocksSinkOptions.builder()
+            .withProperty("jdbc-url", jdbcUrl)
+            .withProperty("load-url", loadUrl)
+            .withProperty("database-name", "test")
+            .withProperty("table-name", "score_board")
+            .withProperty("username", "root")
+            .withProperty("password", "")
+            .build();
+
+    // connector will use a Java object array (Object[]) to represent a row of
+    // StarRocks table, and each element is the value for a column. Need to
+    // define the schema of the Object[] which matches that of StarRocks table
+    TableSchema schema = TableSchema.builder()
+            .field("id", DataTypes.INT().notNull())
+            .field("name", DataTypes.STRING())
+            .field("score", DataTypes.INT())
+            // Must specify the primary key for StarRocks primary key table,
+            // and DataTypes.INT().notNull() for `id` must specify notNull()
+            .primaryKey("id")
+            .build();
+    // Transforms the RowData to the Object[] according to the schema
+    RowDataTransformer transformer = new RowDataTransformer();
+    // Create the sink with schema, options, and transformer
+    SinkFunction<RowData> starRockSink = StarRocksSink.sink(schema, options, transformer);
+    source.addSink(starRockSink);
+    ```
+
+  - The `RowDataTransformer` in the main program is defined as
+
+    ```java
+    private static class RowDataTransformer implements StarRocksSinkRowBuilder<RowData> {
+    
+        /**
+         * Set each element of the object array according to the input RowData.
+         * The schema of the array matches that of StarRocks table.
+         */
+        @Override
+        public void accept(Object[] internalRow, RowData rowData) {
+            internalRow[0] = rowData.id;
+            internalRow[1] = rowData.name;
+            internalRow[2] = rowData.score;
+            // Only need for StarRocks primary key table. Set the last
+            // element to tell whether the record is a UPSERT or DELETE.
+            internalRow[internalRow.length - 1] = StarRocksSinkOP.UPSERT.ordinal();
+        }
+    }  
+    ```
+
+## Best Practices
+
+### Load data to primary key table
+
+This section will show how to load data to StarRocks primary key table to achieve partial update, and conditional update.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/f3dc4b5e-cf05-490a-9e40-f0bb63a5932f)
partial updates and conditional updates